### PR TITLE
Fix: BRGEMM strided when offsets are uniformly spaced apart

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**

### CODE REVIEW

**Comment by chelini** [comment]
Would you mind also adding a test for this case? thanks!

**Review by alheinecke — CHANGES_REQUESTED** [review]
as @chelini already mentioned let's add some unit test. 

The code itself looks good.

**Comment by alheinecke** [comment]
I have also looked at the irdump, and I don't see the pretty print being handled for strided BRGEMM: 

```
 %219 = memref.alloc() : memref<1x1x1x14x32xf32> "res4f_branch2c"
 %220 = arith.addi %arg113, %217 : index "res4f_branch2c"
 xsmm.unary.invoke %219[%c0, %c0, %c0, %c0, %c0] = %145(%100[%c0, %218, %220, %c0, %c0]) : (memref<1x32x14x14x32xf32>) -> memref<1x1x1x14x32xf32> "res4f_branch2c"
 xsmm.brgemm.invoke.f32 "res4f_branch2c"
  %221 = memref.alloc() : memref<1x32x14x14x32xf32> ["res4f_branch2c", "add", "relu"]
 xsmm.binary.invoke %221 = %150(%219, %199) : (memref<1x1x1x14x32xf32>, memref<1x32x14x14x32xf32>) -> memref<1x32x14x14x32xf32> ["res4f_branch2c", "add", "relu"]
 xsmm.unary.invoke %202[%c0, %218, %220, %c0, %c0] = %147(%221[%c0, %218, %220, %c0, %c0]) : (memref<1x32x14x14x32xf32>) -> memref<1x32x14x14x32xf32> ["res4f_branch2c", "add", "relu"]
memref.dealloc %221 : memref<1x32x14x14x32xf32> ["res4f_branch2c", "add", "relu"]
memref.dealloc %219 : memref<1x1x1x14x32xf32> "res4f_branch2c"
```

can you please add?

File: pmlc/dialect/xsmm/ir/ops.cc [hunk 1]
```diff
@@ -274,9 +274,38 @@ static std::string stringfy(BinaryKind kind) {
...
+}
+
+BRGemmInvokeF32Op::operand_range BRGemmInvokeF32Op::getOperandsForC() {
+  auto cType = c().getType().cast<MemRefType>();
+  return getOperands().slice(4, cType.getRank());
+}
 
 void BRGemmInvokeF32Op::print(OpAsmPrinter &p) {
-  // assert(0 && "implement me");
+  auto funcType = FunctionType::get(
```
Comment (chelini) [inline pmlc/dialect/xsmm/ir/ops.cc:297]: why not using the assembly form here? https://mlir.llvm.org/docs/OpDefinitions/#declarative-assembly-format

Comment (alheinecke) [reply pmlc/dialect/xsmm/ir/ops.cc:297]: I guess this fine for now as all the other prints are done in the same way... e.g. when you go to line 157 in the same fi.e

Comment (chelini) [reply pmlc/dialect/xsmm/ir/ops.cc:297]: Fair enough

**Review by alheinecke — APPROVED** [review]
LGTM

**Pull request merged**